llvm/lib/Target/AIE/AIEMachineScheduler.h

```diff
@@ -170,6 +175,11 @@ class AIEPostRASchedStrategy : public PostGenericScheduler {
   // After scheduling a block, fill in nops, apply bundling, etc.
   void commitBlockSchedule(MachineBasicBlock *BB);
 
+  // This function returns true when it is possible to continue
+  // with top-down without entering in loop because all remaining instructions
```

Comment:
nit: infinite loop.

Reply:
Referring to an earlier commit, you say 'all remaining instructions'. Implementing it that way rather than focusing on the N=1 case with a delayslot  would improve readability.